R/latest-dependencies.R

```diff
@@ -0,0 +1,39 @@
+#' Use "latest" versions of all dependencies
+#'
+#' Pins minimum version of dependencies to latest (as determined by "source)"
+#' Useful for tidyverse package, but should otherwise be used with extreme
+#' care.
+#'
+#' @keywords internal
```

Comment:
Re: "extreme care". We could add a `@seealso` to R Packages > Package metadata > Dependencies> Versioning.

> Think carefully if you declare a minimum version. In some sense, the safest thing to do is to require a version greater than or equal to the package’s current version. For public work, this is most naturally defined as the current CRAN version of a package; private or personal projects may adopt some other convention. But it’s important to appreciate the implications for people who try to install your package: if their local installation doesn’t fulfill all of your requirements around versions, installation will either fail or force upgrades of these dependencies. This is desirable if your minimum version requirements are genuine, i.e. your package would be broken otherwise. But if your stated requirements have a less solid rationale, this may be unnecessarily conservative and inconvenient.
>
> In the absence of clear, hard requirements, you should set minimum versions (or not) based on your expected user base, the package versions they are likely to have, and a cost-benefit analysis of being too lax versus too conservative.

Ha! This TODO reveals R Packages should change in response to this PR:

> TODO: return here and insert concrete details closer to publication. Thing 1: mention relevant usethis function(s). Today that is `use_tidy_version()` but it needs some work (https://github.com/r-lib/usethis/issues/771). Thing 2: See if Kirill’s prototype of studying minimum version has matured into anything we can recommend for general use (https://github.com/r-lib/pillar/tree/c017f20476fce431ceee18cc3a637f7ed2884d3a/minver#readme)